test/distribute_reads_test.rb

```diff
@@ -323,18 +332,29 @@ def with_lag(lag)
     end
   end
 
-  def assert_log(message)
+  def prepare_log(proc)
     io = StringIO.new
     previous_logger = DistributeReads.logger
     begin
       DistributeReads.logger = Logger.new(io)
-      yield
+      proc.call
     ensure
       DistributeReads.logger = previous_logger
     end
+
+    io
+  end
+
+  def assert_log(message, &proc)
```

Comment:
Let's use `&block` instead of `&proc` to follow the Ruby convention, and pass it to prepare_log with `&` - `prepare_log(&block)`.